### PR TITLE
Fix PHP 7 Fatal Error when including String.php

### PR DESCRIPTION
**Pull request by alexbowers**

### CODE REVIEW

**Comment by davidpersson** [comment]
Can't we put the check into the String.php file itself? Or does the problem occur when the file is parsed? In that case another option might be to use the new catchable engine exceptions (https://wiki.php.net/rfc/engine_exceptions_for_php7).

```
if (PHP >= 7) {
    return;
}
```

**Comment by alexbowers** [comment]
I honestly didnt even think of that, i'll try it this evening when I get the chance, unless somebody else wants to give it a go.

**Comment by alexbowers** [comment]
@davidpersson Just gave it  a quick go actually, it doesn't work.

image

**Comment by alexbowers** [comment]
As for the catchable interface, since 1.1 isn't PHP 7 exclusive (as far as i'm aware?), should we be using this feature? Would it not cause parse errors in 5.6 etc.

**Comment by jails** [comment]
This should work imo:
```php
class StringDeprecated {
    ...
}
class_alias('StringDeprecated', 'String');
```

**Comment by alexbowers** [comment]
@jails Nope. 
image

**Comment by jails** [comment]
@alexbowers sorry I forgot the `if (PHP_VERSION_ID < 70000)` part.

**Comment by alexbowers** [comment]
Ah okay, i see what you mean then

**Comment by alexbowers** [comment]
Yep, that seems good.

**Comment by alexbowers** [comment]
Changes pushed

File: util/String.php [hunk 1]
```diff
@@ -19,12 +19,16 @@
 $message .= "class with PHP >=7.0.";
 trigger_error($message, E_USER_DEPRECATED);
 
+if(PHP_VERSION_ID < 70000) {
```
Comment (davidpersson) [inline util/String.php]: There is just one space missing here.

Comment (alexbowers) [inline util/String.php]: Fixed

**Comment by alexbowers** [comment]
I assume you dont want all of the comments regarding String::* to be replaced in that class?

**Comment by davidpersson** [comment]
Good catch. But I'd say no we should not do this. Really "Text" should be used and the documentation there is important.

**Comment by alexbowers** [comment]
okay. there is also an unused `use lithium\util\Text` at the top. Do we want this removed or does it not matter?

**Comment by davidpersson** [comment]
It is used actually as far as I can see.

**Comment by alexbowers** [comment]
Weird. PHP storm being weird then.

**Comment by alexbowers** [comment]
Ah, the reason is because they're in the same namespace, no need to 'use' it.

**Comment by davidpersson** [comment]
Yeah, but it's good to be explicit.

**Comment by davidpersson** [comment]
If we're good here, please squash the commits and give it a nice commit message. http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

**Comment by nateabele** [comment]
:+1:

**Comment by alexbowers** [comment]
@davidpersson @nateabele Done.

**Pull request merged**